### DISCUSSION TITLE
Drawing shapes with holes ?

### DISCUSSION THREAD

**Discussion by nyue** [cduck/drawSvg#62] [Q&A] [answered]
I wish to draw a spline shape with e.g. 10 CVs

I wish to then cut a 3 holes with a different sets of CVs inside the first shape

Is this doable with drawSvg ? Is there some example code that illustrate hole cutting ?

I made some progress but I was expecting the triangle to be transparent, it turn out black

```
import drawSvg as draw

d = draw.Drawing(400, 400, origin='center', displayInline=False)

p = draw.Path(stroke='red', stroke_width=2, fill='red')
p.M(20,-20).h(200).v(-160).h(-200).Z()
d.append(p)

p0 = draw.Path(fill_rule='evenodd')
p0.M(80,-40).v(-100).l(100,50).Z()
d.append(p0)

d.saveSvg('test.svg')
d.savePng('test.png')
```

Thank you.

**Comment by cduck** [comment, marked as answer]
Your code is almost there.  Draw both shapes in the same path.
```python
p = draw.Path(stroke='red', stroke_width=2, fill='red', fill_rule='evenodd')
p.M(20,-20).h(200).v(-160).h(-200).Z()
p.M(80,-40).v(-100).l(100,50).Z()
```

**Reply by nyue** [reply]
Thank you. It works.
